resolver/dns/dns_resolver.go

```diff
@@ -51,7 +51,8 @@ const (
 )
 
 var (
-	errMissingAddr = errors.New("missing address")
+	errMissingAddr   = errors.New("dns resolver: missing address")
+	errEndsWithColon = errors.New("dns resolver: address ends with colon")
```

Comment:
"::" also ends with colon but it is a valid case. 
I think more explanation will be better, may be just add a comment above the definition, like `"address ending with a colon that is supposed to be the separator between host and port is not allowed. e.g. "::" is a valid address as it is an IPv6 address(host only) and [::]: is invalid as it ends with a colon as the host and port separator.`

Reply:
Updated error and added comment. PTAL